ci/ci/build.py

```diff
@@ -141,6 +145,12 @@ def __init__(self, params):
             self.deps = [params.name_step[d] for d in json['dependsOn'] if params.name_step[d]]
         else:
             self.deps = []
+        duplicates = [
+            name
+            for name, count in Counter(self.deps).items()
+            if count > 1]
+        if duplicates:
+            raise BuildConfigurationError(f'found duplicate dependencies of {self.name}: {duplicates}')
```

Comment:
I don't think the duplicates are going to be printed out correctly. I think it will be the object hash. You either need to overwrite __str__ for Step or call step.name in duplicates.

```
duplicates = [step.name for step, count in Counter(self.deps).items() if count > 1]